datasets/imagenet_download.py

```diff
@@ -0,0 +1,52 @@
+    tar.close()
+
+def imagenet_prepare_val():
+  # Read in the labels file
+  with open(Path(__file__).parent.parent / "datasets/imagenet/imagenet_2012_validation_synset_labels.txt", 'r') as f:
+    labels = f.read().splitlines()
+  f.close()
+  # Get a list of images
+  images = os.listdir(Path(__file__).parent.parent / "datasets/imagenet/val")
+  images.sort()
+  # Create folders and move files into those
+  for co,dir in enumerate(labels):
+    if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/val" / dir): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/val" / dir)
+    if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/val" / dir / images[co]): os.replace(Path(__file__).parent.parent / "datasets/imagenet/val" / images[co], Path(__file__).parent.parent / "datasets/imagenet/val" / dir / images[co])
+  os.remove(Path(__file__).parent.parent / "datasets/imagenet/imagenet_2012_validation_synset_labels.txt")
+
+def imagenet_prepare_train():
+  images = os.listdir(Path(__file__).parent.parent / "datasets/imagenet/train")
+  for co,tarf in enumerate(images):
+    # for each tar file found. Create a folder with its name. Extract into that folder. Remove tar file
+    if Path(Path(__file__).parent.parent / "datasets/imagenet/train" / images[co]).is_file():
+      images[co] = tarf[:-4] # remove .tar from extracted tar files
+      if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/train" / images[co]): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/train" / images[co])
+      imagenet_extract(Path(__file__).parent.parent / "datasets/imagenet/train" / tarf, Path(__file__).parent.parent / "datasets/imagenet/train" / images[co], small=True)
+      os.remove(Path(__file__).parent.parent / "datasets/imagenet/train" / tarf)
+
+if __name__ == "__main__":
+  if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet"): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet")
+  if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/val"): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/val")
+  if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/train"): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/train")
+  download_file("https://raw.githubusercontent.com/raghakot/keras-vis/master/resources/imagenet_class_index.json", Path(__file__).parent.parent / "datasets/imagenet/imagenet_class_index.json")
+  download_file("https://raw.githubusercontent.com/tensorflow/models/master/research/slim/datasets/imagenet_2012_validation_synset_labels.txt", Path(__file__).parent.parent / "datasets/imagenet/imagenet_2012_validation_synset_labels.txt")
+  download_file("https://image-net.org/data/ILSVRC/2012/ILSVRC2012_img_val.tar", Path(__file__).parent.parent / "datasets/imagenet/ILSVRC2012_img_val.tar") # 7GB
```

Comment:
By default maybe only download the val? And flag to select train also

Reply:
makes sense!